### ISSUE TITLE
OverflowError: signed integer is greater than maximum

### ISSUE THREAD

**Issue by snthibaud** [awslabs/aws-data-wrangler#472]
**Describe the bug**
The exception 'OverflowError: signed integer is greater than maximum' occurred and the following stack trace occurred in production:
```
  File "/listing-service/lib/athena.py", line 21, in query
    result = wr.athena.read_sql_query(q, database=database, ctas_approach=False, boto3_session=boto3_session)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/_config.py", line 361, in wrapper
    return function(**args)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/athena/_read.py", line 730, in read_sql_query
    return _resolve_query_without_cache(
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/athena/_read.py", line 504, in _resolve_query_without_cache
    return _resolve_query_without_cache_regular(
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/athena/_read.py", line 446, in _resolve_query_without_cache_regular
    return _fetch_csv_result(
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/athena/_read.py", line 262, in _fetch_csv_result
    ret = s3.read_csv(
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/s3/_read_text.py", line 260, in read_csv
    return _read_text(
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/s3/_read_text.py", line 138, in _read_text
    ret = _read_text_file(path=paths[0], **args)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/awswrangler/s3/_read_text.py", line 86, in _read_text_file
    df: pd.DataFrame = parser_func(f, **pandas_kwargs)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/pandas/io/parsers.py", line 688, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/pandas/io/parsers.py", line 454, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/pandas/io/parsers.py", line 948, in __init__
    self._make_engine(self.engine)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/pandas/io/parsers.py", line 1180, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/pandas/io/parsers.py", line 2010, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 537, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 711, in pandas._libs.parsers.TextReader._get_header
  File "pandas/_libs/parsers.pyx", line 905, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas/_libs/parsers.pyx", line 2034, in pandas._libs.parsers.raise_parser_error
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/botocore/response.py", line 77, in read
    chunk = self._raw_stream.read(amt)
  File "/opt/conda/envs/listing-service/lib/python3.8/site-packages/urllib3/response.py", line 515, in read
    data = self._fp.read() if not fp_closed else b""
  File "/opt/conda/envs/listing-service/lib/python3.8/http/client.py", line 471, in read
    s = self._safe_read(self.length)
  File "/opt/conda/envs/listing-service/lib/python3.8/http/client.py", line 612, in _safe_read
    data = self.fp.read(amt)
  File "/opt/conda/envs/listing-service/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/opt/conda/envs/listing-service/lib/python3.8/ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "/opt/conda/envs/listing-service/lib/python3.8/ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
```

**To Reproduce**
Steps to reproduce the behavior. Also add details about Python and Wrangler's version and how the library was installed.
I used conda to install awswrangler with the following spec: awswrangler>=1.10.0. I assume the latest conda version at this time was picked (1.10.1).

**Comment by igorborgest** [comment]
Hi @snthibaud, thanks for reporting it.

Do you know what is the column type (Athena/Glue data type) that is raising the issue?

Also, did you try with the default `ctas_approach=True`?

**Comment by snthibaud** [comment]
I can confirm that it is working with CTAS. CTAS might not work for me in all cases, because it does not supported structured data columns, however. I am not sure if a specific column is responsible and am not really sure how to check. It seems to me that the stack trace is more indicative of an issue relating to the size of transferred data over SSL.

**Comment by igorborgest** [comment]
Thanks for reporting it. But I will close this issue because I couldn't replicate it in my environment. Please open a new issue if you find someway to share a reproducible example.

**Comment by jan-xyz** [comment]
@snthibaud could you check if it is fixed if you downgrade your Python to <3.8 ? It seems to be a bug in Python: https://bugs.python.org/issue42853

**Comment by jthetzel** [comment]
@jan-xyz @snthibaud  It appears this bug was introduced in Python 3.8.0 (https://github.com/python/cpython/pull/12698). Upgrading to 3.10.0 likely solves it (according to https://bugs.python.org/msg391359 , but I was not able to test due to other conflicts with 3.10.x). Downgrading to Python 3.7.12 resolved the issue for me.
